cache/cache.go

```diff
@@ -130,7 +119,17 @@ func (c *Cache) Watch(ctx context.Context, key string, opts ...clientv3.OpOption
 				return
 			case <-c.internalCtx.Done():
 				return
-			case events, ok := <-w.eventQueue:
+			case resp, ok := <-w.cancelCh:
+				if !ok {
+					return
+				}
+				select {
+				case <-ctx.Done():
+				case <-c.internalCtx.Done():
+				case responseChan <- resp:
+				}
+				return
+			case resp, ok := <-w.eventsCh:
```

Comment:
There is no deterministic order when selecting on multiple channels. Close on eventsCh can come before event on cancelCh.